modnet/fit_genetic.py

```diff
@@ -48,10 +104,8 @@ def train_val_split(
         Parameter:
             data: 'MODData' data which need to be splitted.
         """
-
-        split_border = int(0.9*data.df_targets.shape[0])
-        final_border = int(data.df_targets.shape[0])
-        self.X_train, self.X_val = data.split((range(split_border),range(split_border,final_border)))
+        i = randint(0,9)
+        self.X_train, self.X_val = self.MDKsplit(data, n_splits=10, random_state=i)[i]
```

Comment:
```suggestion
        self.md_train, self.md_val = self.MDKsplit(data, n_splits=10, random_state=i)[i]
```
and accompanying changes